src/main/java/org/pentaho/di/trans/dataservice/serialization/ServiceTrans.java

```diff
@@ -221,12 +244,38 @@ public TransMeta load( Repository repository ) throws KettleException {
         if ( rd == null ) {
           rd = root; // root
         }
-        return repository.loadTransformation( repository.getTransformationID( name, rd ), null );
+
+        ObjectId transId = repository.getTransformationID( name, rd );
```

Comment:
@mkambol This should have failed earlier at repository.loadRepositoryDirectoryTree(). My guess is that this is caused by dirty env which might have happened when you were playing with reference synchronizer. I don't see a way for reproducing this so for now I'd keep it as is and see if it will happen again and fix root issue instead (REPO_PATH instead of FILE). Moreover as repository.loadRepositoryDirectoryTree() was here before the changes it would fail anyway.